app/controllers/concerns/unconfirmed_user_concern.rb

```diff
@@ -12,7 +12,7 @@ def validate_token
     with_unconfirmed_user do
       result = EmailConfirmationTokenValidator.new(@user).submit
 
-      analytics.track_event(Analytics::EMAIL_CONFIRMATION, result.to_h)
+      analytics.track_event(Analytics::USER_REGISTRATION_EMAIL_CONFIRMATION, result.to_h)
```

Comment:
out of curiosity, what does result.to_h do when it is passed to track_event? Where is that being logged, if at all.

Reply:
This seems to gather additional context which is passed along to Ahoy for tracking, as far as I can tell.

The current code validates the token with `EmailConfirmationTokenValidator`, returning a new `FormResponse` object passed both valid and error response data, together with the user's uuid and whether they're confirmed or not, and finally merged with `to_h` before sending to Ahoy for handling. (So it should all end up in the tracking bucket.)